data/config/schema.xml

```diff
@@ -397,5 +380,104 @@ to save the current settings to XML.
 		<limits min="0" max="6" step="1" />
 		<short>Sort order</short>
 		<long>Currently active sort order</long>
+	</entry>	
+
+	<!-- Webserver Settings -->	
+	<entry name="webserver/access" type="int" value="0">
+		<limits>
+			<enum>Disabled</enum>
+			<enum>localhost</enum>
+			<enum>All Connections</enum>
+			<enum>Custom Netmask</enum>
+		</limits>
+		<short>Webserver access</short>
+		<long>Define if you want to control performous remotely with web interface (requires restart).</long>
+	</entry>
+	<entry name="webserver/netmask" type="int" value="31">
+		<limits>
+		<enum>1</enum>
+		<enum>2</enum>
+		<enum>3</enum>
+		<enum>4</enum>
+		<enum>5</enum>
+		<enum>6</enum>
+		<enum>7</enum>
+		<enum>8</enum>
+		<enum>9</enum>
+		<enum>10</enum>
+		<enum>11</enum>
+		<enum>12</enum>
```

Comment:
If you are going to list enum options, maybe take another step and list the actual mask? Otherwise, couldn't you just use min and max value? Most of the values also don't make sense, e.g. 1 bit mask doesn't do anything useful.

Reply:
1) Using int doesn't work.
2) These aren't bitmasks but slash notation; 1 is /1 or also 128.0.0.0. Arguably you're right that it's not very useful but somebody might still want to use that. Either way, I'm thinking I'll probably leave only 16 to 31 or so in the UI and leave a note in the schema saying if they want more they can edit it directly.

Reply:
Actually I haven't even seen netmask definition with any other webserver, and I've worked with a lot of them. Normally you specify bind addresses and/or hostnames (a name lookup may return multiple addresses to bind to like `::1` and `127.0.0.1` for `localhost`). System services may need to always bind `0.0.0.0` and `::` and then do custom filtering to work with interfaces that may go up and down or change IPs while the service is running, but that is hardly required with a game like Performous. Please try to keep things simple.

As to int type and limits: you should add limits to int types if not currently supported in `configuration.cc`.

Reply:
Ints do support limits but not converting this to a simple user-friendly representation in the UI, which is what I'm doing so far. Anyway, the average user will never have to mess with this (the idea is to have Disabled, localhost, and all connections) just like we do now, but you can also select the custom netmask so you can do things like "only accept connections from WLAN"

Reply:
For that user experience, you probably want a single config option that allows `Internet`, `LAN` and `localhost` only, and looks up the interface address and iface's netmask for the middle option. Another feasible option is to *not* limit by address, always bind `0.0.0.0` and `::` but add an auth token (e.g. included in URL shown as QR code in-game).

Reply:
> For that user experience, you probably want a single config option that allows

The problem is, we have no way to determine from (C++) what constitutes "LAN", so the best bet to support it is to just let the user decide. Again, this is an option that I think is nice to have (and chiefly, because it's pretty easy to implement) but I doubt many people will actually use.

Reply:
Also, both "Custom netmask" and "All Connections" (which is essentially another name for "Internet") will bind to 0.0.0.0; the difference is the former will also make use of RESTinio's IPBlocker feature.